spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -372,7 +373,9 @@ private static Iterator<ManifestFile> buildManifest(
       FileIO io = new HadoopFileIO(conf.get());
       TaskContext ctx = TaskContext.get();
       String suffix =
-          String.format("stage-%d-task-%d-manifest", ctx.stageId(), ctx.taskAttemptId());
+          String.format(
```

Comment:
Why doesn't this cause a "File Already Exists" exception?

Reply:
as we use overwrite when creating Avro writer https://github.com/apache/iceberg/blob/ef7e20e12eae2638015ff91b3356eb2f32f601cc/core/src/main/java/org/apache/iceberg/ManifestListWriter.java#L93

https://github.com/apache/iceberg/blob/ef7e20e12eae2638015ff91b3356eb2f32f601cc/core/src/main/java/org/apache/iceberg/ManifestListWriter.java#L99

I am testing and was thinking of creating a separate PR for this change as I don't think we need overwrite here

Reply:
We do need overwrite as it allows us to bypass some if exists checks in some Hadoop FS implementations.